src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -265,13 +265,17 @@ void IExtensionConfigProvider.Initialize(ExtensionConfigContext context)
             // Throw if any of the configured options are invalid
             this.Options.Validate(this.nameResolver, this.TraceHelper);
 
-            // For 202 support
-            if (this.Options.NotificationUrl == null)
-            {
 #pragma warning disable CS0618 // Type or member is obsolete
-                this.Options.NotificationUrl = context.GetWebhookHandler();
+
+            // Invoke webhook handler to make functions runtime register extension endpoints.
+            context.GetWebhookHandler();
+
+            // This line ensure every time we need the webhook URI, we get it directly from the
+            // function runtime, which has the most up-to-date knowledge about the site hostname.
+            this.HttpApiHandler.RegisterWebhookProvider(
+                this.Options.WebhookUriProviderOverride ??
+                (() => context.GetWebhookHandler()));

```

Comment:
Can this expression be simplified to this?
```suggestion
                context.GetWebhookHandler);
```